src/bootstrap/src/core/config/config.rs

```diff
@@ -1973,6 +1965,43 @@ impl Config {
         config.llvm_plugins = llvm_plugins.unwrap_or(false);
         config.rust_optimize = optimize.unwrap_or(RustOptimize::Bool(true));
 
+        // We make `x86_64-unknown-linux-gnu` use the self-contained linker by default, so we will
+        // build our internal lld and use it as the default linker, by setting the `rust.lld` config
+        // to true by default:
+        // - on the `x86_64-unknown-linux-gnu` target
+        // - on the `dev` and `nightly` channels
+        // - when building our in-tree llvm (i.e. the target has not set an `llvm-config`), so that
+        //   we're also able to build the corresponding lld
+        // - or when using an external llvm that's downloaded from CI, which also contains our prebuilt
+        //   lld
+        // - otherwise, we'd be using an external llvm, and lld would not necessarily available and
+        //   thus, disabled
+        // - similarly, lld will not be built nor used by default when explicitly asked not to, e.g.
+        //   when the config sets `rust.lld = false`
```

Comment:
Having this information here is very useful, but it's also crucial to include it in `config.example.toml`. But duplicating long comments isn't ideal as they can easily fall out of sync. I'm unsure of the best approach to ensure both sides provide this information. Perhaps we could move the detailed information to `config.example.toml` and provide a brief summary here, and refer to `config.example.toml` for further details?

Reply:
@onur-ozkan are you suggesting that I replace what I've added to `config.example.toml` with the above comment?

As for keeping things in sync, a note "if you modify this, please update $otherfile" should likely be enough.

Reply:
Ok, cool.